packages/stock-location/src/services/stock-location.ts

```diff
@@ -172,46 +168,46 @@ export default class StockLocationService extends TransactionBaseService {
    * @param updateData - The update data for the stock location.
    * @returns The updated stock location.
    */
-
+  @InjectEntityManager()
   async update(
     stockLocationId: string,
-    updateData: UpdateStockLocationInput
+    updateData: UpdateStockLocationInput,
+    context: SharedContext = {}
   ): Promise<StockLocation> {
-    return await this.atomicPhase_(async (manager) => {
-      const locationRepo = manager.getRepository(StockLocation)
-
-      const item = await this.retrieve(stockLocationId)
-
-      const { address, ...data } = updateData
-
-      if (address) {
-        if (item.address_id) {
-          await this.updateAddress(item.address_id, address)
-        } else {
-          const locAddressRepo = manager.getRepository(StockLocationAddress)
-          const locAddress = locAddressRepo.create(address)
-          const addressResult = await locAddressRepo.save(locAddress)
-          data.address_id = addressResult.id
-        }
-      }
+    const manager = context.transactionManager!
+    const locationRepo = manager.getRepository(StockLocation)
+
+    const item = await this.retrieve(stockLocationId, undefined, context)
 
-      const { metadata, ...fields } = data
+    const { address, ...data } = updateData
 
-      const toSave = locationRepo.merge(item, fields)
-      if (metadata) {
-        toSave.metadata = setMetadata(toSave, metadata)
+    if (address) {
+      if (item.address_id) {
+        await this.updateAddress(item.address_id, address, context)
+
+        const locAddressRepo = manager.getRepository(StockLocationAddress)
+        const locAddress = locAddressRepo.create(address)
+        const addressResult = await locAddressRepo.save(locAddress)
+        data.address_id = addressResult.id
       }
+    }
```

Comment:
thought: I am not sure the control flow is correct here.

There are two cases to account for:
- address is passed but no address id -> should create a new address and save it to the location
- address is passed alongside address id -> should update the existing address and save it to the location

I believe the latter is not working as expected. It looks like we are creating a new address even though the id is passed. This is why the integration test in the stock location service is failing.

Would something like this work:
```
if (address) {
  if (item.address_id) {
    await this.updateAddress(item.address_id, address, context)
  } else {
    const locAddressRepo = manager.getRepository(StockLocationAddress)
    const locAddress = locAddressRepo.create(address)
    const addressResult = await locAddressRepo.save(locAddress)
    data.address_id = addressResult.id
  }
}
```

This is also similar to what we had before. What motivated you to change that?

Reply:
If this is the intended behaviour, we should change the integration test to reflect that.

Reply:
As a side note (and totally non-blocking), this upsert address flow is duplicated in many places in our code base. One could argue, that we could create a dedicated address module to make our own lives easier.

Reply:
Not sure it is intended, here we update and then we save (which re update + add one read) I think the else disappeared by mistake indeed :) because it was present as far as I recall

Reply:
OMG! This was removed by accident. There is the `else` in it.